mlflow/pytorch/__init__.py

```diff
@@ -67,8 +68,10 @@ def get_default_pip_requirements():
     import torchvision
 
     return [
-        "torch=={}".format(torch.__version__),
-        "torchvision=={}".format(torchvision.__version__),
+        # `torch.__version__` and `torchvision.__version__`contain a local version identifier
+        # (e.g. "cpu" in "1.9.0+cpu") which causes the installation from PyPI to fail.
+        "torch=={}".format(_strip_local_version_identifier(torch.__version__)),
+        "torchvision=={}".format(_strip_local_version_identifier(torchvision.__version__)),
```

Comment:
@dbczumar 
I think `import torch; torch.__version__` is more stable than `importlib_metadata.version`. For example, we could download torch and add its path to `sys.path` (we rarely do this). In this case, `importlib_metadata.version` doesn't work, but `import torch; torch.__version__` does.

Reply:
Other libraries may add local version identifiers in the future, so can we pass all of our version strings through a common function (like you had before) that performs `_strip_local_version_identifier` and can later be extended to perform additional operations?

Reply:
Got it, I'll define `_get_pinned_requirement` (like I did) and run `_strip_local_version_identifier` in it.

Reply:
Done in https://github.com/mlflow/mlflow/pull/4632/commits/5565b54bb6f04d7cb7ebb71d3b0e55c152befed0